lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +37,55 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._storage = {}
+        self.path = path_to_raw_txt_data
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not isinstance(self.path, Path) or not self.path.exists():
+            raise FileNotFoundError
+
+        if not self.path.is_dir():
+            raise NotADirectoryError
+
+        if not self.path.glob("*"):
+            raise EmptyDirectoryError
+
+        meta_list = [elem.name for elem in self.path.glob("*_meta.json")]
+        raw_list = [elem.name for elem in self.path.glob("*_raw.txt")]
+
+        meta_ids = sorted([int(re.search(r'\d+', elem)[0]) for elem in meta_list])
+        raw_ids = sorted([int(re.search(r'\d+', elem)[0]) for elem in raw_list])
+
+        right_list = list(range(1, max(meta_ids[-1], raw_ids[-1])+1))
```

Comment:
how do you check there are no slips and it starts from 1 and finishes at N?

Reply:
here I see list from 1 to whatever is max of id of raw or meta. but it can be 10000 while there are only 100 articles.

Reply:
in this case there are slips (if the max is 10000, but there are only 100 articles), am i wrong? i thought that we're supposed to have a list of ids from 1 to (len(meta/raw)+1) in meta and raw list

Reply:
right and you need to check for it, and raise error if there are slips

Reply:
done